tests/common/mod.rs

```diff
@@ -177,14 +178,26 @@ impl Environment {
         client
     }
 
-    pub fn new_connected_app(&mut self, owner: ClientPublicId) -> TestApp {
-        let mut app = TestApp::new_disconnected(&mut self.rng, owner);
+    pub fn new_connected_app(&mut self, owner: &FullId, perms: Option<AppPermissions>) -> TestApp {
+        let owner_full_id = match owner {
+            FullId::Client(id) => id,
+            _ => panic!("Trying to create app from non client full id"),
+        };
+        let mut app = TestApp::new_disconnected(&mut self.rng, owner_full_id, perms);
         self.establish_connection(&mut app);
         app
     }
 
-    pub fn new_disconnected_app(&mut self, owner: ClientPublicId) -> TestApp {
-        TestApp::new_disconnected(&mut self.rng, owner)
+    pub fn new_disconnected_app(
+        &mut self,
+        owner: &FullId,
```

Comment:
Why did we change the `owner` arg `FullId`? Just curious.

Reply:
Ah, because we need the sign capability for signing the token when we instantiate the app  (at `new_disconnected`)